### PR TITLE
[test](jdbc) add jdbc and hive regression test

### PR DESCRIPTION
**Pull request by morningman**
# Proposed changes

Issue Number: close #xxx

## Problem summary

1. Modify default behavior of `build.sh`
    The `BUILD_JAVA_UDF` is default ON, so that jvm is needed for compilation and runtime.

2. Add docker-compose for MySQL 5.7, PostgreSQL 14 and Hive 2
   See `docker/thirdparties/docker-compose`.

3. Add some regression test cases for jdbc query on MySQL, PG and Hive Catalog
   The default is `false`, if set to true, you need first start docker for MySQL/PG/Hive.

4. Support `if not exists` and `if exists` for create/drop resource and create/drop encryptkey

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes: Need libjvm.so for FE/BE runtime
    - [ ] No
    - [ ] I don't know
6. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
7. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
8. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
9. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

To be fixed.
See `regression-test/suites/jdbc_p0/test_jdbc_query_mysql.groovy` `order_qt_sql2`
`ERROR 1105 (HY000): errCode = 2, detailMessage = ClassCastException: java.time.LocalDateTime cannot be cast to java.sql.Timestamp`

### CODE REVIEW

File: docker/thirdparties/start-thirdparties-docker.sh [hunk 1]
```diff
@@ -20,8 +20,38 @@
...
+sudo rm "${ROOT}"/docker-compose/postgresql/data/data/* -rf
+sudo docker compose -f "${ROOT}"/docker-compose/postgresql/postgresql-14.yaml --env-file "${ROOT}"/docker-compose/postgresql/postgresql-14.env up -d
+
+# hive
+# before start it, you need to download parquet file package, see "README" in "docker-compose/hive/scripts/"
+sed -i "s/doris--/${CONTAINER_UID}/g" "${ROOT}"/docker-compose/hive/hive-2x.yaml
+sed -i "s/doris--/${CONTAINER_UID}/g" "${ROOT}"/docker-compose/hive/hadoop-hive.env.tpl
+sudo docker compose -f "${ROOT}"/docker-compose/hive/hive-2x.yaml --env-file "${ROOT}"/docker-compose/hive/hadoop-hive.env down
+sudo sh "${ROOT}"/docker-compose/hive/gen_env.sh
+sudo docker compose -f "${ROOT}"/docker-compose/hive/hive-2x.yaml --env-file "${ROOT}"/docker-compose/hive/hadoop-hive.env up -d
```
Comment (Ray-Eldath) [inline docker/thirdparties/start-thirdparties-docker.sh:57]: these should be `docker-compose` not `docker compose` I suppose?

File: regression-test/suites/jdbc_p0/test_jdbc_query_pg.groovy [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        String jdbcPg14Table1 = "jdbc_pg_14_table1"
+
+        sql """drop resource if exists $jdbcResourcePg14;"""
+        sql """
+            create external resource $jdbcResourcePg14
+            properties (
+                "type"="jdbc",
+                "user"="postgres",
+                "password"="123456",
+                "jdbc_url"="jdbc:postgresql://127.0.0.1:5442/postgres?currentSchema=doris_test",
```
Comment (Ray-Eldath) [inline regression-test/suites/jdbc_p0/test_jdbc_query_pg.groovy:33]:
```suggestion
                "jdbc_url"="jdbc:postgresql://127.0.0.1:$pg_14_port/postgres?currentSchema=doris_test",
```

File: docker/thirdparties/start-thirdparties-docker.sh [hunk 3]
```diff
@@ -20,8 +20,38 @@
...
+sudo mkdir -p "${ROOT}"/docker-compose/postgresql/data/data
+sudo rm "${ROOT}"/docker-compose/postgresql/data/data/* -rf
+sudo docker compose -f "${ROOT}"/docker-compose/postgresql/postgresql-14.yaml --env-file "${ROOT}"/docker-compose/postgresql/postgresql-14.env up -d
+
+# hive
+# before start it, you need to download parquet file package, see "README" in "docker-compose/hive/scripts/"
+sed -i "s/doris--/${CONTAINER_UID}/g" "${ROOT}"/docker-compose/hive/hive-2x.yaml
+sed -i "s/doris--/${CONTAINER_UID}/g" "${ROOT}"/docker-compose/hive/hadoop-hive.env.tpl
+sudo docker compose -f "${ROOT}"/docker-compose/hive/hive-2x.yaml --env-file "${ROOT}"/docker-compose/hive/hadoop-hive.env down
+sudo sh "${ROOT}"/docker-compose/hive/gen_env.sh
```
Comment (Ray-Eldath) [inline docker/thirdparties/start-thirdparties-docker.sh:56]:
```suggestion
sudo bash "${ROOT}"/docker-compose/hive/gen_env.sh
sudo docker compose -f "${ROOT}"/docker-compose/hive/hive-2x.yaml --env-file "${ROOT}"/docker-compose/hive/hadoop-hive.env down
```

reverse these two lines, then replace `sh` to `bash`.

File: docker/thirdparties/start-thirdparties-docker.sh [hunk 4]
```diff
@@ -20,8 +20,38 @@
...
+sudo docker compose -f "${ROOT}"/docker-compose/postgresql/postgresql-14.yaml --env-file "${ROOT}"/docker-compose/postgresql/postgresql-14.env down
+sudo mkdir -p "${ROOT}"/docker-compose/postgresql/data/data
+sudo rm "${ROOT}"/docker-compose/postgresql/data/data/* -rf
+sudo docker compose -f "${ROOT}"/docker-compose/postgresql/postgresql-14.yaml --env-file "${ROOT}"/docker-compose/postgresql/postgresql-14.env up -d
+
+# hive
+# before start it, you need to download parquet file package, see "README" in "docker-compose/hive/scripts/"
+sed -i "s/doris--/${CONTAINER_UID}/g" "${ROOT}"/docker-compose/hive/hive-2x.yaml
+sed -i "s/doris--/${CONTAINER_UID}/g" "${ROOT}"/docker-compose/hive/hadoop-hive.env.tpl
+sudo sh "${ROOT}"/docker-compose/hive/gen_env.sh
```
Comment (Ray-Eldath) [inline docker/thirdparties/start-thirdparties-docker.sh:55]: this needs to be `bash` not `sh`. I don't know why, but this is the reality.

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**